src/berry_blender.c

```diff
@@ -2077,439 +2236,457 @@ static void Blender_DummiedOutFunc(s16 a0, s16 a1)
-        sUnknown_03000DE8[5] = 0;
+    // Calculate color and feel of pokeblock
+    pokeblock->color = CalculatePokeblockColor(berries, &sPokeblockFlavors[0], numPlayers, numNegatives);
+    sPokeblockFlavors[FLAVOR_COUNT] = (sPokeblockFlavors[FLAVOR_COUNT] / numPlayers) - numPlayers;
 
-    if (pokeblock->color == 12)
+    if (sPokeblockFlavors[FLAVOR_COUNT] < 0)
+        sPokeblockFlavors[FLAVOR_COUNT] = 0;
+
+    if (pokeblock->color == PBLOCK_CLR_BLACK)
     {
-        multiuseVar = Random() % 10;
-        for (i = 0; i < 5; i++)
+        // Black pokeblocks get their flavors randomly reassigned
+        multiuseVar = Random() % ARRAY_COUNT(sBlackPokeblockFlavorFlags);
+        for (i = 0; i < FLAVOR_COUNT; i++)
         {
-            if ((sUnknown_08339CC8[multiuseVar] >> i) & 1)
-                sUnknown_03000DE8[i] = 2;
+            if ((sBlackPokeblockFlavorFlags[multiuseVar] >> i) & 1)
+                sPokeblockFlavors[i] = 2;
             else
-                sUnknown_03000DE8[i] = 0;
+                sPokeblockFlavors[i] = 0;
         }
     }
 
-    for (i = 0; i < 6; i++)
+    for (i = 0; i < FLAVOR_COUNT + 1; i++)
     {
-        if (sUnknown_03000DE8[i] > 255)
-            sUnknown_03000DE8[i] = 255;
+        if (sPokeblockFlavors[i] > 255)
+            sPokeblockFlavors[i] = 255;
     }
 
-    pokeblock->spicy = sUnknown_03000DE8[0];
-    pokeblock->dry = sUnknown_03000DE8[1];
-    pokeblock->sweet = sUnknown_03000DE8[2];
-    pokeblock->bitter = sUnknown_03000DE8[3];
-    pokeblock->sour = sUnknown_03000DE8[4];
-    pokeblock->feel = sUnknown_03000DE8[5];
+    pokeblock->spicy  = sPokeblockFlavors[FLAVOR_SPICY];
+    pokeblock->dry    = sPokeblockFlavors[FLAVOR_DRY];
+    pokeblock->sweet  = sPokeblockFlavors[FLAVOR_SWEET];
+    pokeblock->bitter = sPokeblockFlavors[FLAVOR_BITTER];
+    pokeblock->sour   = sPokeblockFlavors[FLAVOR_SOUR];
+    pokeblock->feel   = sPokeblockFlavors[FLAVOR_COUNT];
 
-    for (i = 0; i < 6; i++)
-    {
-        flavors[i] = sUnknown_03000DE8[i];
-    }
+    for (i = 0; i < FLAVOR_COUNT + 1; i++)
+        flavors[i] = sPokeblockFlavors[i];
 }
 
-static void BlenderDebug_CalculatePokeblock(struct BlenderBerry* berries, struct Pokeblock* pokeblock, u8 playersNo, u8* flavors, u16 a4)
+// Unused
+static void Debug_CalculatePokeblock(struct BlenderBerry* berries, struct Pokeblock* pokeblock, u8 numPlayers, u8* flavors, u16 maxRPM)
 {
-    Blender_CalculatePokeblock(berries, pokeblock, playersNo, flavors, a4);
+    CalculatePokeblock(berries, pokeblock, numPlayers, flavors, maxRPM);
 }
 
-static void sub_8081E20(void)
+static void Debug_SetStageVars(void)
 {
-    u32 frames = (u16)(sBerryBlenderData->gameFrameTime);
-    u16 max_RPM = sBerryBlenderData->max_RPM;
-    s16 var = 0;
+    u32 frames = (u16)(sBerryBlender->gameFrameTime);
+    u16 maxRPM = sBerryBlender->maxRPM;
+    s16 stage = 0;
 
     if (frames < 900)
-        var = 5;
+        stage = 5;
     else if ((u16)(frames - 900) < 600)
```

Comment:
you sure this isn't a range check? (`(frames >= 900 && frames < 1500)`)

Reply:
I did try, I couldn't get it to match that way